src/core/utils.cpp

```diff
@@ -0,0 +1,21 @@
+#include "flockmtl/core/utils.hpp"
+#include <string>
+namespace flockmtl {
+namespace core {
+
+const std::string provider_key = "provider";
+
+static bool helper (const nlohmann::json& json, std::string& provider_name){
```

Comment:
I don't like utils like this. Just put it where it belongs, do:
json.contains(provider_key) ? json.at(provider_key).get<std::string>() : <default_value>

Reply:
I have added comment in discord regarding this

Reply:
done